conan/tools/cmake/layout.py

```diff
@@ -14,13 +14,13 @@ def cmake_layout(conanfile, generator=None, src_folder="."):
         else:
             multi = False
 
-    conanfile.folders.source = src_folder
+    conanfile.folders.source = src_folder if not subfolder else os.path.join(subfolder, src_folder)
```

Comment:
sub folder is above source dir? 😕 

Is that from the perspective of the root folder?